src/app/loadout-builder/process.ts

```diff
@@ -515,17 +496,15 @@ function generateMixesFromPerksOrStats(
 }
 
 function getBaseStatValues(
-  stats: {
-    [index: string]: DimStat;
-  },
   item: DimItem,
   assumeMasterwork: boolean | null,
   lockedModStats: { [statHash: number]: number }
 ) {
+  const stats = _.keyBy(item.stats, (stat) => stat.statHash);
```

Comment:
I'm concerned that moving this in here will lead to it being recomputed when it hasn't changed - I believe it was originally moved out to avoid that.

Reply:
I am not sure how it would have done that. It was created just before the function and passed in. There wasn't any memoizing or anything happening. I will tag the spot where I pulled it from.